### ISSUE TITLE
Undefined names in model_zoo

### ISSUE THREAD

**Issue by cclauss** [dmlc/gluon-cv#100]
flake8 testing of https://github.com/dmlc/gluon-cv on Python 3.6.3

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./gluoncv/model_zoo/cifarresnext.py:122:31: F821 undefined name 'Block'
            total_expansion = Block.expansion ** len(layers)
                              ^
./gluoncv/model_zoo/resnext.py:116:35: F821 undefined name '_conv3x3'
                self.features.add(_conv3x3(64, 1, 0))
                                  ^
./gluoncv/model_zoo/segbase.py:102:21: F821 undefined name '_reshape_like'
            label = _reshape_like(F, label, pred)
                    ^
./gluoncv/model_zoo/segbase.py:102:45: F821 undefined name 'pred'
            label = _reshape_like(F, label, pred)
                                            ^
./gluoncv/model_zoo/segbase.py:103:27: F821 undefined name 'pred'
            loss = -F.sum(pred*label, axis=self._axis, keepdims=True)
                          ^
5     F821 undefined name 'Block'
5
```

**Comment by zhanghang1989** [comment]
What is the advantage of flak8?

**Comment by cclauss** [comment]
I know of only one advantage... __F821 undefined name__  But they are indeed useful.

**Comment by zhanghang1989** [comment]
Yeah. We should probably add flake8
